lib/cocoapods/resolver.rb

```diff
@@ -471,13 +472,25 @@ def spec_is_platform_compatible?(dependency_graph, dependency, spec)
     #         An array of target-appropriate nodes whose `payload`s are
     #         dependencies for `target`.
     #
-    def valid_dependencies_for_target_from_node(target, node)
+    def valid_dependencies_for_target_from_node(target, cached_dependencys, node)
+      cache_dependency = cached_dependencys[node.name]
```

Comment:
`cached_dependencies ||= begin`, that way all the code that does the calculation can be inside the `begin..end` block